velox/functions/lib/Re2Functions.cpp

```diff
@@ -402,16 +424,44 @@ class LikeConstantPattern final : public VectorFunction {
     exec::DecodedArgs decodedArgs(rows, args, context);
     auto toSearch = decodedArgs.at(0);
     if (toSearch->isIdentityMapping()) {
-      auto rawStrings = toSearch->data<StringView>();
-      rows.applyToSelected([&](vector_size_t i) {
-        result.set(i, re2FullMatch(rawStrings[i], re_));
-      });
+      auto input = toSearch->data<StringView>();
+
+      if (fixedPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, likeutils::matchExactPattern(input[i], pattern_));
```

Comment:
matchExactPattern takes input as std::string. Since input[i] is a StringView, calling matchExactPattern requires creating new std::string and copying the data. This code is not necessary. Let's modify the matchExactPattern's signature to access a StringView for input.

Reply:
Thanks for the suggestion, modified accordingly